internal/vds/vds.go

```diff
@@ -171,6 +171,96 @@ func GetAttributeType(attribute string) (int, error) {
 	}
 }
 
+/** Translate C status codes into Go error types */
+func toError(status C.int, ctx *C.Context) error {
+	if status == C.STATUS_OK {
+		return nil
+	}
+
+	msg := C.GoString(C.errmsg(ctx))
+
+	switch status {
+	case C.STATUS_NULLPTR_ERROR: fallthrough
+	case C.STATUS_RUNTIME_ERROR: return NewInternalError(msg)
+	default:
+		return errors.New(msg)
+	}
+}
+
+type RegularSurface struct {
```

Comment:
I think we might want to divide vds.go and vds-test.go files in near future.
But that's after all current refactoring-changes are done.